### ISSUE TITLE
bootstrap application to cluster

### ISSUE THREAD

**Issue by duycuong87vn** [gjovanov/mongo-swarm#3]
hi @gjovanov ,
I see nodejs app on this repo,
Please guide me to add it into the cluster to work with mongo-01, mong-02(routers).
Thanks!

**Comment by gjovanov** [comment]
Hi @duycuong87vn,
the node app was work in progress.

Good news is that I've finished it and if you pull the latest changes, the node app (mongo-log-api) will be deployed either on mongo-01 or mongo-02 VM.

Please note that you will need to recreate the whole Mongo Cluster (with cleanup previous data), as there are some breaking changes due to the Vagrant Group names that don't support - (dash) anymore.

Also, I've done a port forwarding from your Host to the VM's mongo-01 & mongo-02 to access both the API app and the Mongos:

The following ports are available on the Host where you deploy these VMs:

- host port: 8091 mapped to mongo-01:8080 (mongo log API app)
- host port: 8092 mapped to mongo-02:8080 (mongo log API app)
- host port: 27018 mapped to mongo-01:27017
- host port: 27019 mapped to mongo-02:27017

Let me know if you encounter any issue or you have questions.

**Comment by gjovanov** [comment]
Test Write to and Read from the Mongo Cluster
You can test the Write/Read to the Mongo Cluster, you can use the deployed Mongo Log API NodeJS app from the docker repository gjovanov/mongo-log-api, also available in the /app folder.

This API app is deployed in the cluster on VMs mongo-01 and mongo-02 via the same docker stack file: `/ansible/roles/mongo-swarm/docker_stack.yml`

Write
`curl -d '{ }' -H "Content-Type: application/json" -X POST http://localhost:8091/api/log/write`

Read
`curl http://localhost:8091/api/log/read?size=20`

**Comment by duycuong87vn** [comment]
@gjovanov , great,
I can API-Logs deployed to mongo-01 and i can request via curl.

----One problem, I can not connect to mongo-01, mong-02 from outside, 
Although, I checked netstat command and see 
mongo-01, mong-02 Listen on 27018 and 27019. (for me 27019 - 27020)
`d.vm.network "forwarded_port", guest: 27017, host: "#{ i + 1 + 27017 }"`

for more details, please see below.
![swarm_mongo_cluster](https://user-images.githubusercontent.com/6830252/58537468-2355d180-821d-11e9-8832-955b3f334cab.png)
![mong_remote_cnn](https://user-images.githubusercontent.com/6830252/58537505-323c8400-821d-11e9-86f7-d6ce5072f06a.png)
![swar_mong_cluster_iss](https://user-images.githubusercontent.com/6830252/58537570-526c4300-821d-11e9-8ad6-c2e3ebeefcc1.png)
----

Connect to replicaset.
![cnn_to_replicaset](https://user-images.githubusercontent.com/6830252/58538108-9d3a8a80-821e-11e9-8ff5-3a8880b8b990.png)

Primary Node On data-03
![rplc_primary](https://user-images.githubusercontent.com/6830252/58538111-9f9ce480-821e-11e9-8a61-e9e7f1cdaa4d.png)

**Comment by gjovanov** [comment]
Hi @duycuong87vn,
even though it's not recommended approach to exponse access outside the cluster, I've pushed some changes that will enable this.

Pls pull the latest changes and redeploy the mongo stack:
```
user@HOST: vagrant ssh manager-01
vagrant@manager-01: docker stack rm mongo
vagrant@manager-01: cd /vagrant/ansible/roles/mongo-swarm
vagrant@manager-01: docker stack deploy --compose-file docker_stack.yml mongo
```

Also on the HOST check if the subnet masks are old (255.255.255.0) and fix them accordingly to 255.255.0.0:
```
user@HOST: ifconfig
```

If Mask of any vboxnet network is 255.255.255.0 (e.g. vboxnet0), fix it to 255.255.0.0 run this command:

`vboxmanage hostonlyif ipconfig vboxnet0 --ip 10.100.198.1 --netmask 255.255.0.0`

Pls keep an eye on the IP address. In my case vboxnet0 is 10.100.198.1. In your case might be a different one.

After you have made sure the subnets have proper masks, you can connect to mongos from you host like this (assuming u have installed mongo client on your host):
`user@HOST: mongo "mongodb://mongo-01:27017,mongo-02:27017/log"`

**Comment by gjovanov** [comment]
I'm assuming that this one has been fixed due to inactivity.

Feel free to reopen in case it occurs again.
